crates/dht/src/time.rs

```diff
@@ -1075,8 +1037,58 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn compression_all_empty_slices_at_current_time() {
+        let factor = 7;
+        let current_time = Timestamp::now();
+        let store = Arc::new(Kitsune2MemoryOpStore::default());
+
+        let mut pt = PartitionedTime::try_from_store(
+            factor,
+            current_time,
+            store.clone(),
+        )
+        .await
+        .unwrap();
+
+        // Quick calculation of how many full slices we can fit into the current time, leaving space
+        // for the minimum recent time.
+        // Something like 15k full slices at the time of writing.
+        assert_eq!(
+            (current_time.as_micros() as u128 - pt.min_recent_time.as_micros())
+                / (pt.full_slice_duration.as_micros()),
+            pt.full_slices as u128
+        );
+
+        let slice_hash_count = store.slice_hash_count().await.unwrap();
+        // Should be nothing stored, because we haven't ever created any data.
+        assert_eq!(0, slice_hash_count);
+
+        // Getting some partial slice that doesn't have any data stored will just return `None`
+        let some_slice_hash =
+            store.retrieve_slice_hash(5203984823).await.unwrap();
+        assert!(some_slice_hash.is_none());
+
+        // Now insert an op at the current time
+        store
+            .process_incoming_ops(vec![MetaOp {
+                op_id: OpId::from(bytes::Bytes::from(vec![7; 32])),
+                timestamp: Timestamp::now(),
+                op_data: vec![],
+            }])
+            .await
+            .unwrap();
+        // and compute the new state in the future
+        pt.update(Timestamp::now() + 2 * pt.full_slice_duration, store.clone())
+            .await
+            .unwrap();
+
+        // Then a single full slice should have been created
+        assert!(store.slice_hash_count().await.unwrap() > 15_000);
```

Comment:
Store the initial slice hash count at the beginning of the test and then add one?
```suggestion
        assert_eq!(store.slice_hash_count().await.unwrap(),  initial_slice_hash_count + 1);
```

Reply:
Right, so shouldn't this work?
```suggestion
        assert_eq!(store.slice_hash_count().await.unwrap(), initial_pt.full_slices() + 1);
```

Reply:
It does nearly work
- The `full_sices` is calculated on the first update which happens as part of construction. So I just used `pt` here
- It's actually equal and should be. The `slice_hash_count` is meant to report the same value as `full_slices` and that's a valid assertion that I have added
- I do want to keep the original assertion too because that checks that the numbers are sane which checking equality doesn't guarantee

Reply:
Right, definitely keep that assertion.
What the comment before the assertion suggests and what my idea therefore was:
```rust
        let mut pt = PartitionedTime::try_from_store(
            factor,
            current_time,
            store.clone(),
        )
        .await
        .unwrap();
        // Quick calculation of how many full slices we can fit into the current time, leaving space
        // for the minimum recent time.
        // Something like 15k full slices at the time of writing.
        assert_eq!(
            (current_time.as_micros() as u128 - pt.min_recent_time.as_micros())
                / (pt.full_slice_duration.as_micros()),
            pt.full_slices as u128
        );
        let initial_full_slices_count = pt.full_slices;
        ...
        pt.update(Timestamp::now() + 2 * pt.full_slice_duration, store.clone())
            .await
            .unwrap();

        // Then a single full slice should have been created as the current time
        assert!(store.slice_hash_count().await.unwrap() > 15_000);
        assert_eq!(store.slice_hash_count().await.unwrap(), initial_full_slices_count + 1);
```
Does that make sense?

Reply:
Right, yes, I'm being slow. That does make sense. I have to adjust a couple of other bits because I'd added quite a bit of extra time to definitely cover the new op. But I think I now have this set up the way you're after and it does make sense :)